platform/ui/src/components/AllInOneMenu/Menu.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { createContext, ReactNode, useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import './allInOneMenu.css';
+import DividerItem from './DividerItem';
+import PanelSelector from './PanelSelector';
+import classNames from 'classnames';
+import Icon from '../Icon';
+
+export interface MenuProps {
+  className?: string;
+  isVisible?: boolean;
+  preventHideMenu?: boolean;
+  backLabel: string;
+  headerComponent?: ReactNode;
+  showHeaderDivider?: boolean;
+  activePanelIndex?: number;
+  onMenuVisibilityChange?: (isVisible: boolean) => void;
+  children: unknown;
+}
+
+type MenuContextProps = {
+  showSubMenu: (subMenuProps: MenuProps) => void;
+  hideMenu: () => void;
+  addItemPanel: (index: number, label: string) => void;
+  activePanelIndex: number;
+};
+
+type MenuPathState = {
+  props: MenuProps;
+  activePanelIndex: number;
+};
+
+export const MenuContext = createContext<MenuContextProps>(null);
```

Comment:
this probably should be default context not null

Reply:
Well a default context would be weird and not meaningful in this case and according to the official React createContext docs, null is acceptable in cases like this: https://react.dev/reference/react/createContext#parameters.